File: src/app/models/message_full.ts
/* models */
import { Chat } from "./chat";
import { User } from "./user";

export class MessageFull {
  constructor(
    public id: string,
    public chat: Chat,
    public content: string,
    public user: User,
    public createdAt: string,
  ) {}

  public static fromJson(json: any): MessageFull {
    const chat = Chat.fromJson(json.chat);
    const user = User.fromJson(json.user);
    return new MessageFull(
      json.id,
      chat,
      json.content,
      user,
      json.createdAt,
    );
  }
}

File: src-tauri/src/models/message_full.rs
/* system libraries */
use serde::{Deserialize, Serialize};

use super::{chat::Chat, user_data::UserData};

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct MessageFull {
  pub id: String,
  pub chat: Chat,
  pub content: String,
  pub user: UserData,
  pub createdAt: String,
}